server/src/main/resources/org/opensearch/bootstrap/untrusted.policy

```diff
@@ -38,4 +38,6 @@ grant {
   // needed IndyInterface selectMethod (setCallSiteTarget)
   // TODO: clean this up / only give it to engines that really must have it
   permission java.lang.RuntimePermission "getClassLoader";
+  // needed for decay date functions
```

Comment:
This is not needed, please share the issue you have, it should be fixable without policy changes

Reply:
Issue was that painless scripts don't have the required permission to do any system property checks which is happening while checking for featureflag in the parse function of a datetime formatter. Decay date functions like `DecayDateLinear` uses datetime formatters to parse dates at runtime and hence they require the necessary permission. Although if we can agree to remove feature flag entirely then this can be removed. I personally didn't find a good way to give painless the required permission.

Reply:
> Issue was that painless scripts don't have the required permission to do any system property checks which is happening while checking for featureflag in the parse function of a datetime formatter. 

Thank you, I think the painless should not have access to any system property  - this is a restrictive sandbox essentially

Reply:
So, do you see an alternative way to go about this problem considering featureflag is still there?

Reply:
Yes, sure, assuming we keep the feature flag, the feature flags are not dynamic and cannot be changed at runtime, so we could  deduct `canCacheLastParsedFormatter` value from the it.  I believe the only place when we  instantiate `JavaDateFormatter` with possibility of caching is `JavaDateFormatter::combined` method, so we could check feature flag there only once:

```
return new JavaDateFormatter(input, javaPrintFormatter.format, printer, roundUpParsers, parsers, canCacheLastParsedFormatter & FeatureFlags.isEnabled(FeatureFlags.DATETIME_FORMATTER_CACHING));
```

Reply:
Done